packages/affine/block-surface/src/element-model/mindmap.ts

```diff
@@ -0,0 +1,1087 @@
+import type {
+  BaseElementProps,
+  GfxModel,
+  SerializedElement,
+  SurfaceBlockModel,
+} from '@blocksuite/block-std/gfx';
+import type { SerializedXYWH, XYWH } from '@blocksuite/global/utils';
+
+import { LocalConnectorElementModel } from '@blocksuite/affine-model';
+import {
+  GfxGroupLikeElementModel,
+  convert,
+  field,
+  observe,
+  watch,
+} from '@blocksuite/block-std/gfx';
+import {
+  assertType,
+  deserializeXYWH,
+  keys,
+  last,
+  pick,
+} from '@blocksuite/global/utils';
+import { DocCollection, type Y } from '@blocksuite/store';
+import { generateKeyBetween } from 'fractional-indexing';
+import { z } from 'zod';
+
+import type {
+  MindmapNode,
+  MindmapRoot,
+  NodeDetail,
+} from './utils/mindmap/layout.js';
+import type {
+  ConnectorStyle,
+  MindmapStyleGetter,
+} from './utils/mindmap/style.js';
+
+import { TextResizing } from '../consts.js';
+import { ConnectorPathGenerator } from '../managers/connector-manager.js';
+import { LayoutType, layout } from './utils/mindmap/layout.js';
+import {
+  MindmapStyle,
+  applyNodeStyle,
+  mindmapStyleGetters,
+} from './utils/mindmap/style.js';
+
+export { LayoutType } from './utils/mindmap/layout.js';
+export { MindmapStyle } from './utils/mindmap/style.js';
+
+const baseNodeSchema = z.object({
+  text: z.string(),
+  xywh: z.optional(z.string()),
+});
+
+type Node = z.infer<typeof baseNodeSchema> & {
+  children?: Node[];
+};
+
+const nodeSchema: z.ZodType<Node> = baseNodeSchema.extend({
+  children: z.lazy(() => nodeSchema.array()).optional(),
+});
+
+type NodeType = z.infer<typeof nodeSchema>;
+
+function isNodeType(node: Record<string, unknown>): node is NodeType {
+  return typeof node.text === 'string' && Array.isArray(node.children);
+}
+
+export type SerializedMindmapElement = SerializedElement & {
+  children: Record<string, NodeDetail>;
+};
+
+type MindmapElementProps = BaseElementProps & {
+  children: Y.Map<NodeDetail>;
+};
+
+export class MindmapElementModel extends GfxGroupLikeElementModel<MindmapElementProps> {
```

Comment:
I still believe this should be part of the `affine-model`

Reply:
Will address this in further PR.